packages/jest-core/src/testSchedulerHelper.ts

```diff
@@ -13,11 +13,25 @@ const SLOW_TEST_TIME = 1000;
 export function shouldRunInBand(
   tests: Array<Test>,
   timings: Array<number>,
-  {detectOpenHandles, maxWorkers, watch, watchAll}: Config.GlobalConfig,
+  {
+    detectOpenHandles,
+    maxWorkers,
+    watch,
+    watchAll,
+    workerIdleMemoryLimit,
+  }: Config.GlobalConfig,
 ): boolean {
   // detectOpenHandles makes no sense without runInBand, because it cannot detect leaks in workers
   if (detectOpenHandles) {
+    if (workerIdleMemoryLimit) {
+      console.warn(
+        'workerIdleMemoryLimit does not work in combination with detectOpenHandles',
+      );
+    }
```

Comment:
Would be better to validate flags inside `jest-cli` . Like this one:

https://github.com/facebook/jest/blob/75006e46c76f6fda14bbc0548f86edb2ba087cd2/packages/jest-cli/src/cli/args.ts#L17-L20

Reply:
Ah.. Will not work if `workerIdleMemoryLimit` is set in config file.